test/framework/toolchain.py

```diff
@@ -278,16 +278,19 @@ def test_misc_flags_unique(self):
                 tc = self.get_toolchain("goalf", version="1.1.0-no-OFED")
                 tc.set_options({opt: enable})
                 tc.prepare()
+                flag = None
                 if opt == 'optarch':
-                    flag = '-%s' % tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch]
+                    if tc.arch in tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION:
+                        flag = '-%s' % tc.COMPILER_OPTIMAL_ARCHITECTURE_OPTION[tc.arch]
```

Comment:
at the very least, we should make sure that for the major known architectures (mainly x86-64), a matching value for `optarch` *IS* available...

we're making things too liberal here, this may cause us missing an important bug at some time in the future?